frame/election-provider-multi-phase/src/benchmarking.rs

```diff
@@ -227,16 +260,54 @@ frame_benchmarking::benchmarks! {
 		assert!(<MultiPhase<T>>::snapshot().is_some());
 	}
 
+	submit {
+		let c in 1 .. (T::SignedMaxSubmissions::get() - 1);
+
+		// the solution will be worse than all of them meaning the score need to be checked against
+		// ~ log2(c)
+		let solution = RawSolution {
+			score: [(10_000_000u128 - 1).into(), 0, 0],
+			..Default::default()
+		};
+
+		<CurrentPhase<T>>::put(Phase::Signed);
+		<Round<T>>::put(1);
+
+		<SignedSubmissions<T>>::mutate(|queue| {
+			for i in 0..c {
+				let solution = RawSolution {
+					score: [(10_000_000 + i).into(), 0, 0],
+					..Default::default()
+				};
+				let signed_submission = SignedSubmission { solution, ..Default::default() };
+				queue.push(signed_submission);
+			}
+			// as of here, the queue is ordered worst-to-best.
+			// However, we have an invariant that it should be ordered best-to-worst
+			queue.reverse();
```

Comment:
No: a bunch of `push`es followed by `reverse` completes in `O(N)`. A bunch of `insert(0, x)` completes in `O(N**2)`. Inserting at the head of a vector is inefficient enough to want to avoid.